src/CacheManager.StackExchange.Redis/RedisCacheHandle.cs

```diff
@@ -270,6 +270,13 @@ public override bool Exists(string key, string region)
         }
 
         /// <inheritdoc />
+        public override IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keyPattern = GetKey(pattern, region);
+            return _connection.Servers.First().Keys(_redisConfiguration.Database, keyPattern).Select(k => k.ToString());
```

Comment:
Our cluster is mirrored not sharded, so it was working sufficiently in the tests I ran.

For clusters it looks like we will need the key hashset approach then.